src/meta/processors/partsMan/ListHostsProcessor.cpp

```diff
@@ -177,10 +177,34 @@ cpp2::ErrorCode ListHostsProcessor::fillLeaders() {
     HostAddr host;
     TermID term;
     cpp2::ErrorCode code;
+    std::vector<std::string> removeLeadersKey;
     for (; iter->valid(); iter->next()) {
         auto spaceIdAndPartId = MetaServiceUtils::parseLeaderKeyV3(iter->key());
         LOG(INFO) << "show hosts: space = " << spaceIdAndPartId.first
                   << ", part = " << spaceIdAndPartId.second;
+        // If space not exists, remove leader key
+        auto spaceId = spaceIdAndPartId.first;
+        auto partId = spaceIdAndPartId.second;
+
+        // The space id in the leader key may no exist.
+        auto spaceIter = spaceIdNameMap_.find(spaceId);
+        if (spaceIter == spaceIdNameMap_.end()) {
+            removeLeadersKey.emplace_back(iter->key());
+            continue;
+        }
+        // If part invalid , remove leader key
```

Comment:
Maybe just check the space is enough, we don't need to check part again, `drop space` will make sure that all space related will be dropped in an atomic batch. 

Besides, there could be plenty partition, read each part may become a problem.

Reply:
Yeah, the `part ` don't need to be checked.
Because according to the format of the leader key, when the spaceId is valid, the `part `must be valid too.

Good catch, 👍